datafusion/sql/src/planner.rs

```diff
@@ -2837,36 +2861,75 @@ fn remove_join_expressions(
 /// foo = bar => accum=[(foo, bar)] accum_filter=[]
 /// foo = bar AND bar = baz => accum=[(foo, bar), (bar, baz)] accum_filter=[]
 /// foo = bar AND baz > 1 => accum=[(foo, bar)] accum_filter=[baz > 1]
+///
+/// For normal expression join key, assume we have tables -- a(c0, c1 c2) and b(c0, c1, c2):
```

Comment:
I think this kind of join key is typically called an "equijoin" rather than "normal expression join key"

Thank you for these comments